src/main/java/org/jenkinsci/plugins/parallel_test_executor/PreviousTestResultLookup.java

```diff
@@ -0,0 +1,61 @@
+package org.jenkinsci.plugins.parallel_test_executor;
+
+import com.google.common.base.Predicate;
+import hudson.ExtensionPoint;
+import hudson.model.Run;
+import hudson.tasks.test.TestResult;
+import org.jenkinsci.plugins.variant.OptionalExtension;
+import org.jenkinsci.plugins.workflow.cps.nodes.StepStartNode;
+import org.jenkinsci.plugins.workflow.flow.FlowExecution;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+import org.jenkinsci.plugins.workflow.graphanalysis.DepthFirstScanner;
+import org.jenkinsci.plugins.workflow.job.WorkflowRun;
+import org.jenkinsci.plugins.workflow.support.steps.StageStep;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public abstract class PreviousTestResultLookup implements ExtensionPoint {
+    public abstract TestResult lookupTestResult(@Nonnull TestResult result);
+
+    @OptionalExtension(requirePlugins = {"pipeline-stage-step","workflow-cps","workflow-job"})
+    public static class LookupInStage extends PreviousTestResultLookup {
+        @CheckForNull
+        private String stageName;
+
+        public LookupInStage() {
```

Comment:
Apparently unused, and probably unusable. See above.

Reply:
Compiler barfed at me without it.